packages/flutter/test/cupertino/date_picker_test.dart

```diff
@@ -559,25 +559,36 @@ void main() {
         ),
       );
 
-      await tester.drag(find.text('11'), const Offset(0.0, -32.0));
+      // scrolling by this offset will increase the hour column by 1
+      const Offset hourOffset = Offset(0.0, -32.0);
+
+      // 11:59 -> 12:59
+
+      await tester.drag(find.text('11'), hourOffset);
       await tester.pump();
       await tester.pump(const Duration(milliseconds: 500));
 
       expect(date, DateTime(2018, 1, 1, 12, 59));
 
-      await tester.drag(find.text('12'), const Offset(0.0, 32.0));
+      // 12:59 -> 11:59
+
+      await tester.drag(find.text('12'), -hourOffset);
       await tester.pump();
       await tester.pump(const Duration(milliseconds: 500));
 
       expect(date, DateTime(2018, 1, 1, 11, 59));
 
-      await tester.drag(find.text('11'), const Offset(0.0, 64.0));
+      // 11:59 -> 9:59
+
+      await tester.drag(find.text('11'), -hourOffset * 2);
       await tester.pump();
       await tester.pump(const Duration(milliseconds: 500));
 
       expect(date, DateTime(2018, 1, 1, 9, 59));
 
-      await tester.drag(find.text('9'), const Offset(0.0, -192.0));
+      // 9:59 -> 15:59
+
+      await tester.drag(find.text('9'), hourOffset * 6);
```

Comment:
Can you please add a new test to show that the bug you pointed out isn't occurring anymore? i.e. start at 5h. Change the AM/PM value, assert that it's now 17h. Move the hour by one. Make sure it's now 18h and not 6h.

Reply:
On it

Reply:
Added in last commit